src/main/java/uk/gov/laa/gpfd/config/JwtAuthenticationFilter.java

```
@@ -17,17 +18,76 @@
 public class JwtAuthenticationFilter extends OncePerRequestFilter {
     private static final String TOKEN_PREFIX = "bearer ";
     private static final int TOKEN_PARTS = 3;
+    private static final String JWT_PAYLOAD_TENANT_ID_KEY = "tid";
+    private static final String JWT_PAYLOAD_APPLICATION_ID_KEY = "appid";
+    private static final String SCOPE_KEY = "scp";
+    private static final String SCOPE_VALUE = "User.Read";
+
+    private final JwtDecoder jwtDecoder;
+    private final AppConfig appConfig;
+
+    public JwtAuthenticationFilter(JwtDecoder jwtDecoder, AppConfig appConfig) {
+        this.jwtDecoder = jwtDecoder;
+        this.appConfig = appConfig;
+    }
 
     @Override
     public void doFilterInternal(HttpServletRequest servletRequest, @NotNull HttpServletResponse servletResponse, @NotNull FilterChain filterChain) throws IOException, ServletException {
         var token = servletRequest.getHeader("Authorization");
 
-        if (token != null && !token.isEmpty()) {
-            // JWT validation logic to come in later PR
+        if (token == null || token.isEmpty()) {
+            filterChain.doFilter(servletRequest, servletResponse);
+        } else {
+            validateJwt(token);
+            filterChain.doFilter(servletRequest, servletResponse);
         }
+    }
+
+    public boolean validateJwt(String token) {
```

The current `validateJwt()` method returns a boolean, but it always throws exceptions on failure rather than returning `false`. This can be misleading since callers might expect a `false` return for invalid tokens

---

void return type reduces testability of method
returning false would require return of error message, throwing exception fails fast
happy to leave as is, may return false in future iterations